composer/metrics/nlp.py

```diff
@@ -381,14 +446,31 @@ def update(self, batch: dict, output_logits: torch.Tensor, labels: torch.Tensor)
             perplexity = torch.exp(cross_entropy)
             perplexities.append(perplexity)
 
+        metric_result_dict = copy.deepcopy(self.metric_result_dict)
         for (start, end), gold_idx in zip(batch['choice_groupings'], batch['gold_indices']):
             subset = perplexities[start:end]
             idx_min = subset.index(min(subset))
-
             if idx_min == gold_idx:
                 self.correct += torch.tensor(1.0)
+                metric_result_dict['result'].append(1)
+            else:
+                metric_result_dict['result'].append(0)
+
+            question = batch['input_ids'][start][:batch['continuation_indices'][start][0]]
+
+            # TODO: Seems broken for schema tasks
```

Comment:
We might not be setting our continuation indices correctly for schema tasks but needs more investigation

Reply:
Not broken - schema tasks' continuation_ids are actually constant. The 'context' changes and the 'continuation' stays the same, and we take the item with a lower perplexity and the model's 'choice'. Thus, continuation was always the same for these tasks and I was getting spooked. I added more logged content (the entirety of MC options, ie the context + answer combined for all possible answers) to make this clear.